### PR TITLE
Separated types checkers  for sponsored objects

### PR DESCRIPTION
**Pull request by rokuz**

### CODE REVIEW

File: indexer/ftypes_sponsored.hpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+class IsBookingChecker : public BaseSponsoredChecker
+{
+  IsBookingChecker() : BaseSponsoredChecker("booking") {}
+public:
+  DECLARE_CHECKER_INSTANCE(IsBookingChecker);
+};
+
+class IsOpentableChecker : public BaseSponsoredChecker
```
Comment (ygorshenin) [inline indexer/ftypes_sponsored.hpp]: Looks like you may also wrap declarations of these small checkers into macro :)

Comment (milchakov) [reply indexer/ftypes_sponsored.hpp]: +1

**Comment by rokuz** [comment]
DUDE JTTP

File: indexer/ftypes_sponsored.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace ftypes
+{
+class BaseSponsoredChecker : public BaseChecker
+{
+protected:
+  explicit BaseSponsoredChecker(std::string const & sponsoredType);
+};
+
+#define SPONSORED_CHECKER(ClassName, Caterory) \
```
Comment (ygorshenin) [inline indexer/ftypes_sponsored.hpp]: Category

File: indexer/ftypes_sponsored.hpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  class ClassName : public BaseSponsoredChecker { \
+  ClassName() : BaseSponsoredChecker(Caterory) {} \
+  public: DECLARE_CHECKER_INSTANCE(ClassName); };
+
+// Checkers.
+SPONSORED_CHECKER(IsBookingChecker, "booking");
+SPONSORED_CHECKER(IsOpentableChecker, "opentable");
+SPONSORED_CHECKER(IsViatorChecker, "viator");
+SPONSORED_CHECKER(IsThorChecker, "thor");
```
Comment (ygorshenin) [inline indexer/ftypes_sponsored.hpp]:
```c++
#undef SPONSORED_CHECKER
```

Comment (rokuz) [reply indexer/ftypes_sponsored.hpp]: done

**Pull request merged**